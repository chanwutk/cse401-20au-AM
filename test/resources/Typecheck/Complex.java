class Complex {
	public static void main(String[] args) {
		System.out.println(new C1().f(0));
	}
}

class C1 extends C2 {
	int f;
	C2 x;
	public int f(int x) {
		C2 y;
		y = this;
		return new C2().f(x);
	}
}

class C2 {
	C1 x;
	public int f(int i) {
		C2 y;
		try {
			y = x;
		} catch (RuntimeException e) {}
		return x.f(i);
	}
}
