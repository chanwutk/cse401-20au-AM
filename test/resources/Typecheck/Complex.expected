CLASS C1
    VAR f: int
    VAR x: C2
    METHOD f(int): int
        VAR x: int
        VAR y: C2
CLASS Complex
CLASS C2
    VAR x: C1
    METHOD f(int): int
        VAR i: int
        VAR y: C2
